tests/rbd/rbd_snapshot_rename_advanced.py

```diff
@@ -0,0 +1,214 @@
+        **kw: test data
+    """
+    try:
+        pool = kw["config"][pool_type]["pool"]
+        image = kw["config"][pool_type]["image"]
+        snap = kw["config"][pool_type].get("snap", f"{pool}_{image}_snap")
+        clone = kw["config"][pool_type].get("clone", f"{pool}_{image}_clone")
+        size = kw["config"][pool_type]["size"]
+        size_int = int(re.findall(r"\d+", size)[0])
+        snap_rename = kw["config"][pool_type].get(
+            "snap_rename", f"{pool}_{image}_new_snap"
+        )
+        snap_name = f"{pool}/{image}@{snap}"
+
+        if rbd.snap_create(pool, image, snap):
+            log.error(f"Creation of snapshot {snap} failed")
+            return 1
+
+        if rbd.protect_snapshot(snap_name):
+            log.error(f"Snapshot protect failed for {snap_name}")
+            return 1
+
+        log.info("Create a clone of the original image using one of the snaps")
+        snap_name = f"{pool}/{image}@{snap}"
+        if rbd.create_clone(snap_name, pool, clone):
+            log.error(f"Creation of clone {clone} failed")
+            return 1
+
+        resize_int = size_int + 2
+        resize = f"{resize_int}{size.replace(str(size_int), '')}"
+
+        results = {}
+
+        log.info("Rename the snapshot when IO on Clone is in progress. (use rbd bench)")
+        with parallel() as p:
+            image_spec = f"{pool}/{clone}"
+            p.spawn(
+                execute,
+                rbd,
+                "benchwrite",
+                results,
+                imagespec=image_spec,
+            )
+            p.spawn(
+                execute,
+                rbd,
+                "snap_rename",
+                results,
+                pool_name=pool,
+                image_name=image,
+                current_snap_name=snap,
+                new_snap_name=snap_rename,
+            )
+
+        if results["snap_rename"] or results["benchwrite"]:
+            log.error(f"Renaming snapshot while writing IOs failed for {snap_name}")
+            return 1
+
+        log.info(
+            "Rename the snapshot, when resize operation is happening on Clone image."
+        )
+        with parallel() as p:
+            p.spawn(
+                execute,
+                rbd,
+                "image_resize",
+                results,
+                pool_name=pool,
+                image_name=clone,
+                size=resize,
+            )
+            p.spawn(
+                execute,
+                rbd,
+                "snap_rename",
+                results,
+                pool_name=pool,
+                image_name=image,
+                current_snap_name=snap_rename,
+                new_snap_name=f"{snap_rename}_1",
+            )
+
+        if results["snap_rename"] or results["image_resize"]:
+            log.error(
+                f"Renaming snapshot while failed for {snap_name} while performing resize on clone {clone}"
```

Comment:
'while' is misplaced in sentence, please correct

Reply:
Done.